asmcomp/compilenv.ml

```diff
@@ -205,6 +208,11 @@ let get_global_info global_ident = (
             let (ui, crc) = read_unit_info filename in
             if ui.ui_name <> modname then
               raise(Error(Illegal_renaming(modname, ui.ui_name, filename)));
+            (match ui.ui_for_pack, current_unit.ui_for_pack with
```

Comment:
Can you add a comment explaining why this check isn't symmetric (between [ui.ui_for_pack] and [current_unit.ui_for_pack])?